src/endpoint/s3/ops/s3_put_bucket_lifecycle.js

```diff
@@ -94,6 +95,13 @@ async function put_bucket_lifecycle(req) {
             current_rule.id = uuid();
         }
 
+        // Check for duplicate ID in the rules
+        if (id_set.has(current_rule.id)) {
+            dbg.error('Rule ID must be unique. Found same ID for more than one rule: ', current_rule.id);
+            throw new S3Error({ ...S3Error.InvalidArgument, message: 'Rule ID must be unique. Found same ID for more than one rule'});
```

Comment:
I see that you use the same string in the logs and in the message, you can move it to a variable and reuse it if you want.

Reply:
I don't think we need to do that, as only here we are using this error message and it can be modified accordingly.